lib/ansible/plugins/action/win_updates.py

```diff
@@ -103,27 +104,19 @@ def _merge_dict(self, original, new):
 
     def _execute_module_with_become(self, module_name, module_args, task_vars,
                                     wrap_async, use_task):
-        orig_become = self._play_context.become
-        orig_become_method = self._play_context.become_method
-        orig_become_user = self._play_context.become_user
-
-        if not use_task:
-            if orig_become is None or orig_become is False:
-                self._play_context.become = True
-            if orig_become_method != 'runas':
-                self._play_context.become_method = 'runas'
-            if orig_become_user is None or orig_become_user == 'root':
-                self._play_context.become_user = 'SYSTEM'
-
+        orig_become = self._connection.become
         try:
+            if not use_task and orig_become is None:
+                become = become_loader.get('runas')
+                become.set_options(direct={'become_user': 'SYSTEM', 'become_pass': None})
+                self._connection.set_become_plugin(become)
```

Comment:
I don't understand the windows become method well, but I'm confused as to why this is necessary?  Shouldn't it just work with what the user provides?  This seems like confusing implicit behavior.

Reply:
The win_updates module is a complicated beast where we either need to run with become or as a scheduled task for it to be able to access the Windows Update API. The Action plugin pretty much forces become to be used with the `SYSTEM`, like root, account if become is not already set so user's don't have to do so themselves.

They still have the ability to set their own become account which is why we check on `orig_become` or use the older and more error prone scheduled task option with `use_task`.